doc/Developing.md

```diff
@@ -276,6 +277,25 @@ If you have accidentally installed the wrong `yarn` version, you have to remove
 [all-in-one packages]: https://github.com/felixrieseberg/windows-build-tools
 [bug]: https://github.com/theia-ide/theia/issues
 
+### macOS
+
+You need to have the xcode command line tools installed in order to build and run Theia. You can install the tools by running
+
+    xcode-select --install
+ 
+If you receive the following error when running
+
+    Error: spawn node-gyp ENOENT
```

Comment:
@daveyc123 What version of npm, node and yarn do you use?

Reply:
@simark @hexa00 Could it be related to https://github.com/theia-ide/theia/blob/af29995c1dc3f95e6a45ecc88a864704dad8df0a/packages/terminal/package.json#L11? That node-gyp is not pulled transitively because of it?

Reply:
node-gyp is not required by xterm but by @theia/process that depends on node-pty which needs it to build, so the v3 should not be the issue

Reply:
https://github.com/angular/angular-cli/issues/3639#issuecomment-269951156 - `node-gyp` is pulled via `electron-rebuild`, the proper fix is to use at least node 7.9.0